core/src/main/scala/org/apache/spark/executor/CoarseGrainedExecutorBackend.scala

```diff
@@ -201,7 +202,8 @@ private[spark] object CoarseGrainedExecutorBackend extends Logging {
         clientMode = true)
       val driver = fetcher.setupEndpointRefByURI(driverUrl)
       val cfg = driver.askSync[SparkAppConfig](RetrieveSparkAppConfig(executorId))
-      val props = cfg.sparkProperties ++ Seq[(String, String)](("spark.app.id", appId))
+      val props = cfg.sparkProperties ++ Seq[(String, String)](("spark.app.id", appId),
+        ("spark.metrics.namespace", metricsNamespace))
```

Comment:
Why can't the driver provide this as it would with any other Spark property? i.e. if I set `spark.metrics.namespace` in my SparkConf then shouldn't that be what's used here?